### ISSUE TITLE
Unable to reproduce the results in the paper

### ISSUE THREAD

**Issue by Riser6** [TyroneLi/CUA_O3D#3]
I followed the script you provided to extract features from the training scenes, and then performed 3D Distillation Training using the script from Step 4. However, when I proceeded to evaluation in Step 5, the results I obtained were quite different from those reported in your paper.

Specifically, for the 2D fusion model, the mIoU and accuracy are 0.5198 and 0.6270, respectively. But for the 3D distillation model, the mIoU and accuracy dropped significantly to 0.4160 and 0.5334. This is far below the numbers reported in your paper.

The major issue is that many categories have scores of zero. Below are the detailed per-class results for both the 2D fusion and 3D distillation models:
2d fusion:
wall          : 0.724   (8451318/11671074)
floor         : 0.821   (7401167/9018835)
cabinet       : 0.450   (800349/1776695)
bed           : 0.741   (720533/971892)
chair         : 0.602   (2421215/4020943)
sofa          : 0.554   (576971/1042377)
table         : 0.494   (1005462/2033799)
door          : 0.525   (1025700/1955026)
window        : 0.544   (847955/1558767)
bookshelf     : 0.687   (895415/1302507)
picture       : 0.274   ( 74879/273592)
counter       : 0.374   (109095/291869)
desk          : 0.421   (399118/948091)
curtain       : 0.568   (468483/824605)
refrigerator  : 0.511   (134897/264045)
shower curtain: 0.000   (     0/154356)
toilet        : 0.686   ( 95768/139548)
sink          : 0.480   ( 64697/134748)
bathtub       : 0.628   ( 62528/99642 )
otherfurniture: 0.312   (763329/2446436)
Mean IoU 0.5198071422494009
Mean Acc 0.6270680225085365

3d distill:
wall          : 0.720   (9480620/13164719)
floor         : 0.746   (7033263/9430666)
cabinet       : 0.426   (728286/1710247)
bed           : 0.648   (719821/1110548)
chair         : 0.586   (2598372/4435142)
sofa          : 0.641   (604395/942348)
table         : 0.324   (1204158/3715767)
door          : 0.482   (989488/2054828)
window        : 0.521   (822648/1577992)
bookshelf     : 0.520   (733626/1411929)
picture       : 0.000   (    24/216466)
counter       : 0.422   (128415/304257)
desk          : 0.000   (     0/655163)
curtain       : 0.552   (493819/894049)
refrigerator  : 0.401   (107532/267830)
shower curtain: 0.000   (     9/159418)
toilet        : 0.665   ( 83645/125758)
sink          : 0.445   ( 52094/117173)
bathtub       : 0.000   (     0/96355 )
otherfurniture: 0.221   (651116/2940122)
Mean IoU 0.4160470223760598
Mean Acc 0.53341130675740

However, I used your provided script to generate LSeg-DINO-SD features on the ScanNet training set, and for the ScanNet validation set, I directly downloaded the LSeg features provided by OpenScene and moved them into the current project. From what I’ve seen in the feature extraction code, these seem to be consistent, so I don’t think this should affect the experimental results.

In addition, during training, the best IoU obtained in the per-epoch evaluation was only 0.366. I’m wondering if I might have done something wrong during this process that caused me not to achieve the expected results.

I would greatly appreciate it if you could provide me with any suggestions.

**Comment by TyroneLi** [comment]
> I followed the script you provided to extract features from the training scenes, and then performed 3D Distillation Training using the script from Step 4. However, when I proceeded to evaluation in Step 5, the results I obtained were quite different from those reported in your paper.
> 
> Specifically, for the 2D fusion model, the mIoU and accuracy are 0.5198 and 0.6270, respectively. But for the 3D distillation model, the mIoU and accuracy dropped significantly to 0.4160 and 0.5334. This is far below the numbers reported in your paper.
> 
> The major issue is that many categories have scores of zero. Below are the detailed per-class results for both the 2D fusion and 3D distillation models: 2d fusion: wall : 0.724 (8451318/11671074) floor : 0.821 (7401167/9018835) cabinet : 0.450 (800349/1776695) bed : 0.741 (720533/971892) chair : 0.602 (2421215/4020943) sofa : 0.554 (576971/1042377) table : 0.494 (1005462/2033799) door : 0.525 (1025700/1955026) window : 0.544 (847955/1558767) bookshelf : 0.687 (895415/1302507) picture : 0.274 ( 74879/273592) counter : 0.374 (109095/291869) desk : 0.421 (399118/948091) curtain : 0.568 (468483/824605) refrigerator : 0.511 (134897/264045) shower curtain: 0.000 ( 0/154356) toilet : 0.686 ( 95768/139548) sink : 0.480 ( 64697/134748) bathtub : 0.628 ( 62528/99642 ) otherfurniture: 0.312 (763329/2446436) Mean IoU 0.5198071422494009 Mean Acc 0.6270680225085365
> 
> 3d distill: wall : 0.720 (9480620/13164719) floor : 0.746 (7033263/9430666) cabinet : 0.426 (728286/1710247) bed : 0.648 (719821/1110548) chair : 0.586 (2598372/4435142) sofa : 0.641 (604395/942348) table : 0.324 (1204158/3715767) door : 0.482 (989488/2054828) window : 0.521 (822648/1577992) bookshelf : 0.520 (733626/1411929) picture : 0.000 ( 24/216466) counter : 0.422 (128415/304257) desk : 0.000 ( 0/655163) curtain : 0.552 (493819/894049) refrigerator : 0.401 (107532/267830) shower curtain: 0.000 ( 9/159418) toilet : 0.665 ( 83645/125758) sink : 0.445 ( 52094/117173) bathtub : 0.000 ( 0/96355 ) otherfurniture: 0.221 (651116/2940122) Mean IoU 0.4160470223760598 Mean Acc 0.53341130675740
> 
> However, I used your provided script to generate LSeg-DINO-SD features on the ScanNet training set, and for the ScanNet validation set, I directly downloaded the LSeg features provided by OpenScene and moved them into the current project. From what I’ve seen in the feature extraction code, these seem to be consistent, so I don’t think this should affect the experimental results.
> 
> In addition, during training, the best IoU obtained in the per-epoch evaluation was only 0.366. I’m wondering if I might have done something wrong during this process that caused me not to achieve the expected results.
> 
> I would greatly appreciate it if you could provide me with any suggestions.


Hi after looking over your evaluation details, I think there should be something that does not match the preprocessing or training phase, you can first take a look over here for my evaluation details before the submission. These two days (after ddl), I will upload my processed 2D Lseg feature embeddings for you to refer to, and you can then re-extract DINOv2 and SD to re-perform the training.
You can find my logs here first: https://drive.google.com/drive/folders/19IuSypv6yscbceVNP2VFCL4LTzPqm1Az?usp=sharing.

**Comment by TyroneLi** [comment]
> I followed the script you provided to extract features from the training scenes, and then performed 3D Distillation Training using the script from Step 4. However, when I proceeded to evaluation in Step 5, the results I obtained were quite different from those reported in your paper.
> 
> Specifically, for the 2D fusion model, the mIoU and accuracy are 0.5198 and 0.6270, respectively. But for the 3D distillation model, the mIoU and accuracy dropped significantly to 0.4160 and 0.5334. This is far below the numbers reported in your paper.
> 
> The major issue is that many categories have scores of zero. Below are the detailed per-class results for both the 2D fusion and 3D distillation models: 2d fusion: wall : 0.724 (8451318/11671074) floor : 0.821 (7401167/9018835) cabinet : 0.450 (800349/1776695) bed : 0.741 (720533/971892) chair : 0.602 (2421215/4020943) sofa : 0.554 (576971/1042377) table : 0.494 (1005462/2033799) door : 0.525 (1025700/1955026) window : 0.544 (847955/1558767) bookshelf : 0.687 (895415/1302507) picture : 0.274 ( 74879/273592) counter : 0.374 (109095/291869) desk : 0.421 (399118/948091) curtain : 0.568 (468483/824605) refrigerator : 0.511 (134897/264045) shower curtain: 0.000 ( 0/154356) toilet : 0.686 ( 95768/139548) sink : 0.480 ( 64697/134748) bathtub : 0.628 ( 62528/99642 ) otherfurniture: 0.312 (763329/2446436) Mean IoU 0.5198071422494009 Mean Acc 0.6270680225085365
> 
> 3d distill: wall : 0.720 (9480620/13164719) floor : 0.746 (7033263/9430666) cabinet : 0.426 (728286/1710247) bed : 0.648 (719821/1110548) chair : 0.586 (2598372/4435142) sofa : 0.641 (604395/942348) table : 0.324 (1204158/3715767) door : 0.482 (989488/2054828) window : 0.521 (822648/1577992) bookshelf : 0.520 (733626/1411929) picture : 0.000 ( 24/216466) counter : 0.422 (128415/304257) desk : 0.000 ( 0/655163) curtain : 0.552 (493819/894049) refrigerator : 0.401 (107532/267830) shower curtain: 0.000 ( 9/159418) toilet : 0.665 ( 83645/125758) sink : 0.445 ( 52094/117173) bathtub : 0.000 ( 0/96355 ) otherfurniture: 0.221 (651116/2940122) Mean IoU 0.4160470223760598 Mean Acc 0.53341130675740
> 
> However, I used your provided script to generate LSeg-DINO-SD features on the ScanNet training set, and for the ScanNet validation set, I directly downloaded the LSeg features provided by OpenScene and moved them into the current project. From what I’ve seen in the feature extraction code, these seem to be consistent, so I don’t think this should affect the experimental results.
> 
> In addition, during training, the best IoU obtained in the per-epoch evaluation was only 0.366. I’m wondering if I might have done something wrong during this process that caused me not to achieve the expected results.
> 
> I would greatly appreciate it if you could provide me with any suggestions.

Hi, I have uploaded my processed 2D projected Lseg files under here: https://drive.google.com/drive/folders/19IuSypv6yscbceVNP2VFCL4LTzPqm1Az?usp=sharing, you can refer to  released_reExtract_scannet_lseg folder and use our points index to re-extract DINOv2 and Stable Diffusion embeddings as some randomness may possibly happen during the embedding extraction, then re-distill. Meanwhile, you can also find our corresponding model checkpoints.
